lib/builders/gpu-builder.ts

```diff
@@ -0,0 +1,171 @@
+     */
+    nodeRole?: iam.Role,
+    /** 
+     * Optional, AMI Type for Windows Nodes
+     */
+    windowsAmiType?: NodegroupAmiType,
+    /** 
+     * Optional, Desired number of nodes to use for the cluster. 
+     */
+    desiredNodeSize?: number,
+    /** 
+     * Optional, Minimum number of nodes to use for the cluster. 
+     */
+    minNodeSize?: number,
+    /** 
+     * Optional, Maximum number of nodes to use for the cluster. 
+     */
+    maxNodeSize?: number,
+    /** 
+     * Optional, Block device size.
+     */
+    blockDeviceSize?: number,
+    /**
+     * Optional, Cluster Provider Tags.
+     */
+    clusterProviderTags?: {
+        [key: string]: string;
+    },
+
+    /**
+     * Optional, Node Group Tags for AL2 nodes 
+     * which run standard cluster software.
+     */
+    nodeGroupTags?: {
+        [key: string]: string;
+    }
+}
+
+/**
+ * Default props to be used when creating the GPU nodes 
+ * for EKS cluster
+ */
+const defaultOptions: GpuOptions = {
+    kubernetesVersion: eks.KubernetesVersion.of("1.27"),
+    instanceClass: ec2.InstanceClass.G5,
+    instanceSize: ec2.InstanceSize.XLARGE,
+    nodeRole: blueprints.getNamedResource("node-role") as iam.Role,
+    windowsAmiType: NodegroupAmiType.AL2_X86_64_GPU,
+    desiredNodeSize: 2,
+    minNodeSize: 2,
+    maxNodeSize: 3,
+    blockDeviceSize: 50,
+    clusterProviderTags: {
+        "Name": "blueprints-gpu-eks-cluster",
+        "Type": "generic-gpu-cluster"
+    },
+    nodeGroupTags: {
+        "Name": "Mng-linux",
+        "Type": "Managed-linux-Node-Group",
+        "LaunchTemplate": "Linux-Launch-Template",
+    }
+  };
+export class GpuBuilder extends blueprints.BlueprintBuilder {
+
+     /**
+     * This method helps you prepare a blueprint for setting up windows nodes with 
+     * usage tracking addon
+     */
+     public static builder(options: GpuOptions): GpuBuilder {
+        const builder = new GpuBuilder();
+        const mergedOptions = merge(defaultOptions, options);
+
+        builder
+            .clusterProvider(
+                new blueprints.GenericClusterProvider({
+                    version: mergedOptions.kubernetesVersion,
+                    tags: mergedOptions.clusterProviderTags,
+                    role: blueprints.getResource(context => {
+                        return new iam.Role(context.scope, 'ClusterRole', { 
+                            assumedBy: new iam.ServicePrincipal("eks.amazonaws.com"),
+                            managedPolicies: [
+                                iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonEKSClusterPolicy"),
+                                iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonEKSVPCResourceController")
+                            ]
+                        });
+                    }),
+                    managedNodeGroups: [
+                        addNodeGroup(mergedOptions)
+                    ]
+                })
+            )
+            .addOns(
```

Comment:
I think you should have GPU Addon as part of the builder. You should also be adding other required addons for the cluster which you used in your sample runs like coredns, kube-proxy etc etc to minimally run the solution. You can follow observability builder as an example to have a method like `enableGPU` and have all addons. Also as i mentioned before, you should think about grabbing all values as parameters from pattern you are building to builder class to pass to GPU Addon like AmpAddon.